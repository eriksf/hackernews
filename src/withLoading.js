import React, { Component } from 'react'
import Loading from './Loading'

const withLoading = (Component) => ({ isLoading, ...rest }) =>
    isLoading
        ? <Loading />
        : <Component { ...rest } />

export default withLoading
