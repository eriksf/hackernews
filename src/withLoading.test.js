import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import withLoading from './withLoading'
import Button from './Button'

const ButtonWithLoading = withLoading(Button)

describe('withLoading', () => {
    const props = {
        isLoading: false,
        onClick: () => {},
    }

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ButtonWithLoading {...props}>More</ButtonWithLoading>, div)
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(<ButtonWithLoading {...props}>More</ButtonWithLoading>)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
