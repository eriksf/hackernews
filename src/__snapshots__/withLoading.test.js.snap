// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading has a valid snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  More
</button>
`;
